src/rtl_sdr.c

```diff
@@ -165,6 +192,14 @@ int main(int argc, char **argv)
 		}
 	}
 
+	// for now only input to output in same format, input CS16 to all, and CS12 to CS16
+	if (input_format != output_format
+			&& input_format != SOAPY_SDR_CS16
+			&& (input_format != SOAPY_SDR_CS12 || output_format != SOAPY_SDR_CS16)) {
```

Comment:
`input_format != SOAPY_SDR_CS16` since these are strings, this should be `strcmp() == 0` Technically if the format pointer came from one of the format `#defines` (and not user input) its probably use the same pointer to the string constant in BSS, but seems a little dangerous in my opinion.